src/nvim/api/buffer.c

```diff
@@ -663,6 +672,222 @@ void nvim_buf_set_lines(uint64_t channel_id,
   try_end(err);
 }
 
+/// Sets (replaces) a range in the buffer.
+///
+/// Indexing is zero-based, end-exclusive.
+///
+/// To insert text at a given index, set `start` and `end` ranges to the same
+/// index. To delete a range, set `replacement` to an empty array.
+///
+/// Prefer nvim_buf_set_lines when modifying entire lines.
```

Comment:
I should probably note that it retains extmarks here

Reply:
Both functions retain extmarks?